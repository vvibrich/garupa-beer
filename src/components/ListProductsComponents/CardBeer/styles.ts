import { StyleSheet  } from 'react-native';
export const style = StyleSheet.create({
    card: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        borderRadius: 15
    },
    view: {
        flexDirection: 'row',
        justifyContent:'space-between'
    },
    image: {
        width: 100,
        height: 400
    },
    productContainer: {
        flexDirection: 'column',
        marginLeft: 10,
        marginRight: 10
    },
    productNameContainer: {
        flexDirection: 'row',
        justifyContent:'space-between',
        paddingRight: 300
    },
    productNameText: {
        color: '#FFB900'
    },
    productTagline: {
        flexDirection: 'row',
        justifyContent:'space-between',
        paddingRight: 300
    },
    badgeFirstBrewed: {
        flexDirection: 'row',
        justifyContent:'space-between',
        paddingRight: 200,
        marginTop: 20
    },
    containerTap: {
        flexDirection: 'row',
        justifyContent:'space-between',
        paddingRight: 200,
        marginTop: 20
    },
    tapText: {
        width: 190,
        height: 35,
        backgroundColor: "#FFB900",
        textAlign: 'center',
        paddingTop: 7,
        fontWeight: '700',
        color: '#FFF',
        borderRadius: 4
    },
})