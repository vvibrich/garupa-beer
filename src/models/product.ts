interface Volume {
    value: number;
    unit: string;
}

interface Temp {
    value: number;
    unit: string;
}

interface Malt {
    name: string;
    amount: Volume;
}

interface Hop {
    name: string;
    amount: Volume;
    add: string;
    attribute: string;
}

interface Method {
    mash_temp: {
        temp: Temp;
        duration: number;
    }[];
    fermentation: {
        temp: Temp;
    };
    twist: null | string;
}

interface Ingredients {
    malt: Malt[];
    hops: Hop[];
    yeast: string;
}

export interface Product {
    id: number;
    name: string;
    tagline: string;
    first_brewed: string;
    description: string;
    image_url: string;
    abv: number;
    ibu: number;
    target_fg: number;
    target_og: number;
    ebc: number;
    srm: number;
    ph: number;
    attenuation_level: number;
    volume: Volume;
    boil_volume: Volume;
    method: Method;
    ingredients: Ingredients;
    food_pairing: string[];
    brewers_tips: string;
    contributed_by: string;
}