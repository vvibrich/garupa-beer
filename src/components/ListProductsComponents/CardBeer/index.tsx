import React from 'react';
import { TouchableOpacity, View, Image } from 'react-native';
import { Text, Card, Badge } from '@rneui/themed';
import { Product } from '../../../models/product';
import { NavigationProp, ParamListBase, useNavigation } from "@react-navigation/native";
import { style } from './styles';

interface ProductDetailProps {
    product: Product;
}

const CardBeer = ({ product }: ProductDetailProps) => {

    const navigation = useNavigation<NavigationProp<ParamListBase>>();

    return (
        <Card key={product.id} containerStyle={style.card}>
            <TouchableOpacity onPress={() => navigation.navigate("Details", {
                product: product
            })}>
                <View style={style.view}>
                    <Image source={{ uri: product.image_url }} style={style.image} />
                    <View style={style.productContainer}>
                        <View style={style.productNameContainer}>
                            <Text h2 style={style.productNameText}>{product.name}</Text>
                        </View>
                        <View style={style.productTagline}>
                            <Text h3>{product.tagline}</Text>
                        </View>
                        <View style={style.badgeFirstBrewed}>
                            <Badge value={`First Brewed in ${product.first_brewed}`} status='warning' />
                        </View>
                        <View style={style.containerTap}>
                            <Text style={style.tapText}>
                                Tap on card to see more
                            </Text>
                        </View>
                    </View>
                </View>
            </TouchableOpacity>
        </Card>
    );
}



export default CardBeer;