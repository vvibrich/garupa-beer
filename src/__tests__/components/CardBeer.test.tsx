import React from 'react';
import renderer from 'react-test-renderer';
import CardBeer from '../../components/ListProductsComponents/CardBeer';
import { product } from '../../mocks/product';

const mockProduct = product;

jest.mock('@react-navigation/native', () => ({
    useNavigation: () => ({ navigate: jest.fn() }),
}));

test('renders correctly', () => {
    const tree = renderer.create(
        <CardBeer product={mockProduct} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
