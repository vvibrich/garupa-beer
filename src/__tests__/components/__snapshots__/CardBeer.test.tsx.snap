// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderRadius": 15,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "transparent",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://images.punkapi.com/v2/keg.png",
            }
          }
          style={
            {
              "height": 400,
              "width": 100,
            }
          }
        />
        <View
          style={
            {
              "flexDirection": "column",
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingRight": 300,
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#FFB900",
                  "fontSize": 43.714285714285715,
                }
              }
            >
              Buzz
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingRight": 300,
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#242424",
                  "fontSize": 36,
                }
              }
            >
              A Real Bitter Experience.
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 20,
                "paddingRight": 200,
              }
            }
          >
            <View
              style={{}}
              testID="RNE__Badge__Container"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#faad14",
                    "borderColor": "#fff",
                    "borderRadius": 9,
                    "borderWidth": 0.5,
                    "height": 18,
                    "justifyContent": "center",
                    "minWidth": 18,
                  }
                }
                testID="RNE__Badge"
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 12,
                      "paddingHorizontal": 4,
                    }
                  }
                >
                  First Brewed in 09/2007
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 20,
                "paddingRight": 200,
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "backgroundColor": "#FFB900",
                  "borderRadius": 4,
                  "color": "#FFF",
                  "fontWeight": "700",
                  "height": 35,
                  "paddingTop": 7,
                  "textAlign": "center",
                  "width": 190,
                }
              }
            >
              Tap on card to see more
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
